### PR TITLE
Upload analysis config file before submitting jobs

### LINKED ISSUE #58: Analysis config doesn't get uploaded if only 1 job is launched
**Describe the bug**

If only 1 job is launched (e.g. when launching `protopipe-SUBMIT_JOBS` with `-test True`) the output directory on DIRAC doesn't get created until its completion.
This causes the config file upload (which happens immediately after job submission) to crash.

**To Reproduce**

launch any data job (TRAINING, DL2) with only 1 file

**Expected behavior**

The script shouldn't crash at config submission.

**Proposed solution**

Move this step before job submission

### PR DESCRIPTION
**Pull request by HealthyPear**
Fixes #58 

## Caveats

- list of SEs is hardcoded (see #66)

### CODE REVIEW

File: protopipe_grid_interface/scripts/submit_jobs.py [hunk 1]
```diff
@@ -417,6 +417,34 @@ def main():
     log.debug("Particle type: %s", particle)
     log.debug("Energy estimation: %s", estimate_energy)
 
+    # Upload analysis configuration file for provenance
+    if switches["upload_analysis_cfg"]:
+        se_list = ["CC-IN2P3-USER", "DESY-ZN-USER", "CNAF-USER", "CEA-USER"]
```
Comment (kosack) [inline protopipe_grid_interface/scripts/submit_jobs.py:422]: shouldn't this come from `grid.conf` rather than be hard-coded?

Comment (HealthyPear) [reply protopipe_grid_interface/scripts/submit_jobs.py:422]: I opened issue #58 about this (what is`grid.conf`?)

Comment (kosack) [reply protopipe_grid_interface/scripts/submit_jobs.py:422]: I mean the list of sites is hard-coded here, but it could change.  Anyhow, could be a separate issue

File: protopipe_grid_interface/scripts/submit_jobs.py [hunk 2]
```diff
@@ -417,6 +417,34 @@ def main():
     log.debug("Particle type: %s", particle)
     log.debug("Energy estimation: %s", estimate_energy)
 
+    # Upload analysis configuration file for provenance
+    if switches["upload_analysis_cfg"]:
```
[the remark this answers is not in the document]
Comment (kosack) [reply protopipe_grid_interface/scripts/submit_jobs.py:421]: as long as the user knows what dry means, I don't think it's any different. I just have trouble reading long bits of code with nested IFs, since it's hard to follow.   It would be better to just have an info at the top saying 
```py3
if switches['dry']:
   log.info("Dry run enabled. No files will be uploaded")
```

**Review by kosack — CHANGES_REQUESTED** [review]

File: protopipe_grid_interface/scripts/submit_jobs.py (hunk 1 above)
Comment (HealthyPear) [reply protopipe_grid_interface/scripts/submit_jobs.py:422]: Sorry I meant #66

Comment (HealthyPear) [reply protopipe_grid_interface/scripts/submit_jobs.py:422]: though we could use that: we place this list there and the `analysis_metadata.yaml` already pinpoints its location in the analysis folder so it should work - I'll update the issue

File: protopipe_grid_interface/scripts/submit_jobs.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (HealthyPear) [reply protopipe_grid_interface/scripts/submit_jobs.py:421]: OK this might be better

File: protopipe_grid_interface/scripts/submit_jobs.py (hunk 1 above)
Comment (HealthyPear) [reply protopipe_grid_interface/scripts/submit_jobs.py:422]: actually not, I'll open a new PR has it will change also at least another script

**Review by kosack — APPROVED** [review]
